File: Frontend/ParsingPass/test/dev/vale/parsing/AfterRegionsTests.scala
package dev.vale.parsing

import dev.vale.lexing.{BadExpressionEnd, BadStartOfStatementError, ForgotSetKeyword}
import dev.vale.parsing.ast._
import dev.vale.{Collector, StrI}
import org.scalatest._

class AfterRegionsTests extends FunSuite with Collector with TestParseUtils {

  // This test does not pass yet, use #[ignore].
  test("Forgetting set when changing") {
    val error =
      compileStatement(
        """ship.x = 4;""".stripMargin).expectErr()
    error match {
      case ForgotSetKeyword(_) =>
    }
  }

  // This test does not pass yet, use #[ignore].
  test("Report leaving out semicolon or ending body after expression, for paren") {
    compileBlockContents(
      """
        |  a = 3;
        |  set x = 7 )
        """.stripMargin).expectErr() match {
      case BadStartOfStatementError(_) =>
    }
  }

}

File: src/parsing/tests/after_regions_tests.rs
use crate::keywords::Keywords;
use crate::lexing::errors::ParseError;
use crate::parse_arena::ParseArena;
use crate::parsing::tests::utils::{compile_block_contents, compile_statement};
use bumpalo::Bump;

#[test]
fn forgetting_set_when_changing() {
  // This test does not pass yet, use #[ignore].
  let parse_bump = Bump::new();
  let parse_arena = ParseArena::new(&parse_bump);
  let keywords = Keywords::new_for_parse(&parse_arena);
  let error = compile_statement(&parse_arena, &keywords, "ship.x = 4;").unwrap_err();
  match error {
    ParseError::ForgotSetKeyword(_) => {}
    other => panic!("Expected ForgotSetKeyword, got {:?}", other),
  }
}
